modules/ml/src/main/java/org/apache/ignite/ml/svm/SVMLinearBinaryClassificationTrainer.java

```diff
@@ -32,33 +35,50 @@
  * and +1 labels for two classes and makes binary classification. </p> The paper about this algorithm could be found
  * here https://arxiv.org/abs/1409.1458.
  */
-public class SVMLinearBinaryClassificationTrainer implements Trainer<SVMLinearBinaryClassificationModel, LabeledDataset> {
+public class SVMLinearBinaryClassificationTrainer<K, V>
+    implements DatasetTrainer<K, V, SVMLinearBinaryClassificationModel> {
     /** Amount of outer SDCA algorithm iterations. */
-    private int amountOfIterations = 20;
+    private int amountOfIterations = 200;
 
     /** Amount of local SDCA algorithm iterations. */
-    private int amountOfLocIterations = 50;
+    private int amountOfLocIterations = 100;
 
     /** Regularization parameter. */
-    private double lambda = 0.2;
+    private double lambda = 0.4;
 
-    /** This flag enables distributed mode for this algorithm. */
-    private boolean isDistributed;
+    /** Dataset. */
+    private Dataset<SVMPartitionContext, LabeledDataset<Double, LabeledVector>> dataset;
 
     /**
-     * Returns model based on data
+     * Trains model based on the specified data.
      *
-     * @param data data to build model
-     * @return model
+     * @param datasetBuilder   Dataset builder.
+     * @param featureExtractor Feature extractor.
+     * @param lbExtractor      Label extractor.
+     * @param cols             Number of columns.
+     * @return Model.
      */
-    @Override public SVMLinearBinaryClassificationModel train(LabeledDataset data) {
-        isDistributed = data.isDistributed();
+    @Override public SVMLinearBinaryClassificationModel fit(DatasetBuilder<K, V> datasetBuilder, IgniteBiFunction<K, V, double[]> featureExtractor, IgniteBiFunction<K, V, Double> lbExtractor, int cols) {
 
-        final int weightVectorSizeWithIntercept = data.colSize() + 1;
+        assert datasetBuilder != null;
+
+        PartitionDataBuilder<K, V, SVMPartitionContext, LabeledDataset<Double, LabeledVector>> partDataBuilder = new SVMPartitionDataBuilderOnHeap<>(
+            featureExtractor,
+            lbExtractor,
+            cols
+        );
+
+        this.dataset = datasetBuilder.build(
```

Comment:
Dataset is `AutoCloseable` object, so after `fit` it would be great to close it to clear resources. Try-with-resources is preferable.